File: Cpp/1523_count_odd_numbers_in_an_interval_range.cpp
// https://leetcode.cn/problems/count-odd-numbers-in-an-interval-range/

#include <iostream>

using namespace std;

class Solution {
public:
    int countOdds(int low, int high) {
        int res = (high - low + 1) / 2;
        if (low % 2 && high % 2) res += 1;
        return res;
    }
};

int main() {
    Solution sol;
    int low = 3, high = 7;
    cout << sol.countOdds(low, high) << '\n';
    return 0;
}

File: Python3/1523_count_odd_numbers_in_an_interval_range.py
# https://leetcode.cn/problems/count-odd-numbers-in-an-interval-range/

class Solution:
    def countOdds(self, low: int, high: int) -> int:
        res = (high - low + 1) // 2
        if low % 2 and high % 2:
            res += 1
        return res